cmd/init.go

```
@@ -257,146 +274,152 @@ func (cmd *InitCmd) Run(cobraCmd *cobra.Command, args []string) {
-		})
-
-		if selectedOption == useDevSpaceCloud {
-			cmd.useCloud = true
-		} else if selectedOption == useDevSpaceCloudOwnCluster {
-			cmd.useCloud = true
-			connectCluster = true
+		registries, err := provider.GetRegistries()
+		if err != nil {
+			return "", fmt.Errorf("Error retrieving registries: %v", err)
+		}
+		if len(registries) > 0 {
+			registryURL = registries[0].URL
 		} else {
-			cmd.useCloud = false
+			registryURL = "hub.docker.com"
 		}
 	}
 
-	// Check if DevSpace Cloud should be used
-	if cmd.useCloud == false {
-		cmd.configureCluster()
-	} else {
-		// Configure cloud provider
-		cmd.providerName = ptr.String(cloudconfig.DevSpaceCloudProviderName)
-
-		// Choose cloud provider
-		if providerConfig.Default != "" {
-			cmd.providerName = &providerConfig.Default
-		} else if len(providerConfig.Providers) > 1 {
-			options := []string{}
-			for _, provider := range providerConfig.Providers {
-				options = append(options, provider.Name)
-			}
+	return registryURL, nil
 
-			cmd.providerName = ptr.String(survey.Question(&survey.QuestionOptions{
-				Question: "Select a cloud provider",
-				Options:  options,
-			}))
-		}
+}
 
-		// Ensure user is logged in
-		err = cloud.EnsureLoggedIn(providerConfig, *cmd.providerName, log.GetInstance())
-		if err != nil {
-			log.Fatal(err)
-		}
+func (cmd *InitCmd) getRegistryURL(cloudRegistryHostname string) string {
+	useDockerHub := "Use " + dockerHubHostname
+	useDevSpaceRegistry := "Use " + cloudRegistryHostname + " (free, private Docker registry)"
+	useOtherRegistry := "Use other registry"
+	registryUsernameHint := " => you are logged in as %s"
+	registryDefaultOption := useDevSpaceRegistry
+	registryLoginHint := "Please login via `docker login%s` and try again."
 
-		// Create generated yaml if cloud
-		generatedConfig, err := generated.LoadConfig()
-		if err != nil {
-			log.Fatal(err)
+	config := configutil.GetConfig()
+	dockerClient, err := docker.NewClient(config, false, log.GetInstance())
```

This should throw an error because something with docker is messed up. If this fails it does not mean docker is not installed it usually means there are some wrong environment variables set and we should inform the user about this because it will lead to an error later on anyways

---

But would this return an error if Docker is not installed or not?

---

Done